drivers/usbc/tcpc/stm32_ucpd.c

```diff
@@ -0,0 +1,1426 @@
+/*
+ * Copyright 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_ucpd
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ucpd_stm32gx, CONFIG_USBC_LOG_LEVEL);
+
+#include <device.h>
+#include <sys/util.h>
+#include <kernel.h>
+#include <soc.h>
+#include <stddef.h>
+#include <math/ilog2.h>
+
+#include "stm32_ucpd_priv.h"
+
+/**
+ * @brief Zephyr has difficulty with shared interrupt. This flag is used to
+ *        configure a shared interrupt once and is set to false afterwards.
+ */
+static bool irq_config = true;
+static void config_tcpc_irq(void);
+
+/**
+ * @brief UCPD TX ORDSET values
+ */
+static int ucpd_txorderset[] = {
+	/* SOP ORDSET */
+	LL_UCPD_ORDERED_SET_SOP,
+	/* SOP PRIME ORDSET */
+	LL_UCPD_ORDERED_SET_SOP1,
+	/* SOP PRIME PRIME ORDSET */
+	LL_UCPD_ORDERED_SET_SOP2,
+	/* SOP PRIME DEBUG ORDSET */
+	LL_UCPD_ORDERED_SET_SOP1_DEBUG,
+	/* SOP PRIME PRIME DEBUG ORDSET */
+	LL_UCPD_ORDERED_SET_SOP2_DEBUG,
+	/* HARD RESET ORDSET */
+	LL_UCPD_ORDERED_SET_HARD_RESET,
+	/* CABLE RESET ORDSET */
+	LL_UCPD_ORDERED_SET_CABLE_RESET,
+};
+
+/**
+ * @brief Test for a goodCRC message
+ *
+ * @returns true if message is goodCRC, else false
+ */
+static bool ucpd_msg_is_good_crc(union pd_header header)
+{
+	/*
+	 * Good CRC is a control message (no data objects) with GOOD_CRC
+	 * message type in the header.
+	 */
+	return (header.number_of_data_objects == 0 &&
+		header.extended == 0 &&
+		header.message_type == PD_CTRL_GOOD_CRC);
+}
+
+/**
+ * @brief Apply the UCPD CC1 and CC2 pin configurations.
+ *
+ *        UCPDx_STROBE: UCPDx pull-down configuration strobe:
+ *                      when UCPDx is enabled, with CC1 and CC2 pin UCPD
+ *                      control bits configured: apply that configuration.
+ */
+static void update_stm32g0x_cc_line(UCPD_TypeDef *ucpd_port)
+{
+	if ((uint32_t)(ucpd_port) == UCPD1_BASE) {
+		SYSCFG->CFGR1 |= SYSCFG_CFGR1_UCPD1_STROBE_Msk;
+	} else {
+		SYSCFG->CFGR1 |= SYSCFG_CFGR1_UCPD2_STROBE_Msk;
```

Comment:
This function is compiled out for other series because its use is guarded:

if (IS_ENABLED(CONFIG_SOC_SERIES_STM32G0X)) {
                update_stm32g0x_cc_line(config->ucpd_port);
}

Reply:
cf feedback on previous remark

Reply:
I changed the IS_ENABLED to #ifdef